scripts/update-browser-releases/bun.ts

```diff
@@ -0,0 +1,347 @@
+  iso ? new Date(iso).toISOString().split('T')[0] : undefined;
+
+/**
+ * Extracts the version number from a GitHub release.
+ * @param r - The GitHub release object.
+ * @returns The version number, or undefined if not found.
+ */
+const extractVersion = (r: GitHubRelease): string | undefined => {
+  const hay = `${r.tag_name ?? ''} ${r.name ?? ''}`;
+  const m = hay.match(SEMVER_RE);
+  return m ? m[1] : undefined;
+};
+
+/**
+ * Fetches all Bun GitHub releases across pages.
+ * @returns A list of GitHub release objects.
+ */
+const fetchAllReleases = async (): Promise<GitHubRelease[]> => {
+  const all: GitHubRelease[] = [];
+  for (let page = 1; page <= 10; page++) {
+    const res = await fetch(GITHUB_API + page, {
+      headers: {
+        'User-Agent':
+          'MDN-Browser-Release-Update-Bot/1.0 (+https://developer.mozilla.org/)',
+        Accept: 'application/vnd.github+json',
+      },
+    });
+    if (!res.ok) {
+      throw new Error(`GitHub releases fetch failed: HTTP ${res.status}`);
+    }
+    const chunk = (await res.json()) as GitHubRelease[];
+    if (!Array.isArray(chunk) || chunk.length === 0) {
+      break;
+    }
+    all.push(...chunk);
+  }
+  return all;
+};
+
+/**
+ * Resolves the Bun release asset suffix for the current platform/arch.
+ * @returns The asset suffix like "darwin-aarch64" or "linux-x64".
+ */
+const resolveAssetSuffix = () => {
+  const plat = process.platform === 'darwin' ? 'darwin' : process.platform;
+  const cpu = process.arch === 'arm64' ? 'aarch64' : process.arch;
+  return `${plat}-${cpu}`;
+};
+
+/**
+ * Recursively searches for the Bun binary within an extracted directory.
+ * @param dir The directory to search within.
+ * @returns The path to the Bun binary, or null if not found.
+ */
+const findBunBinaryPath = async (dir: string): Promise<string | null> => {
+  const entries = await fs.readdir(dir, { withFileTypes: true });
+  for (const e of entries) {
+    const p = path.join(dir, e.name);
+    if (e.isDirectory()) {
+      const found = await findBunBinaryPath(p);
+      if (found) {
+        return found;
+      }
+    } else if (e.isFile() && e.name === 'bun') {
+      return p;
+    }
+  }
+  return null;
+};
+
+/**
+ * Downloads and extracts a specific Bun version for the current platform.
+ * Uses a local cache and reuses artifacts when available.
+ * @param version The Bun version to download, for example "1.2.21".
+ * @returns The absolute path to the extracted Bun binary, or null on failure.
+ */
+const downloadAndExtractBun = async (
+  version: string,
+): Promise<string | null> => {
+  const suffix = resolveAssetSuffix();
+  const base = `https://github.com/oven-sh/bun/releases/download/bun-v${version}`;
+  const asset = `bun-${suffix}.zip`;
+  const url = `${base}/${asset}`;
+
+  const cacheRoot = path.join(process.cwd(), '.cache', 'bun-updater', version);
+  const zipPath = path.join(cacheRoot, asset);
+  const extractDir = path.join(cacheRoot, 'extracted');
```

Comment:
It doesn't look like we're cleaning up the `extracted` dir, so we should probably use a unique folder for each release?

```suggestion
  const extractDir = path.join(cacheRoot, suffix);
```